internal/core/src/index/VectorMemIndex.cpp

```diff
@@ -600,6 +600,10 @@ VectorMemIndex<T>::Query(const DatasetPtr dataset,
                                       search_conf[RANGE_FILTER],
                                       GetMetricType());
             }
+            // `range_search_k` is only used as one of the conditions for early termination.
```

Comment:
you can use existed param "k" for range search

Reply:
The meanings are not quite the same
1. `range_search_k` does not promise to return exactly `k` results.
2. `k` use `uint`, but `range_search_k` wanna use `-1` to preserve the possibility of returning all results without k limitation.

Reply:
ok